### PR TITLE
Acrn boot 2

### PR DESCRIPTION
**Pull request by jren1**
Those patches will:
1) allow the grub to pass the kernel cmdline to ACRN sos kernel
2) enable USB HID (keyboard, mouse) and ethernet in SOS

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -297,7 +297,7 @@ ifneq ($(BUILD),local)
 	@docker build --build-arg GOVER=$(GOVER) --build-arg USER=$(USER) --build-arg GROUP=$(GROUP) \
                       --build-arg UID=$(UID) --build-arg GID=$(GID) \
                       $(DOCKER_HTTP_PROXY) $(DOCKER_HTTPS_PROXY) $(DOCKER_NO_PROXY) $(DOCKER_ALL_PROXY) \
-                      -t $@ build-tools/src/scripts > /dev/null
+                      -t $@ build-tools/src/scripts
```
Comment (rvs) [inline Makefile]: Any particular reason you're making it more verbose by removing >/dev/null ?

I mean -- I suppose we need to introduce a full verbose mode with something like make V=1 ... but that would be a separate PR.

Comment (jren1) [reply Makefile]: sorry, That is for debugging purpose only. I will change back

Comment (jren1) [reply Makefile]: Solved, and re-pushed the patches in the same PR 354. Please review again

**Comment by jren1** [comment]
@rvs 
>Another question is this: what's the process for the patches you're adding to be merged into acrn and acrn's linux kernel upstream? IOW, when can we expect https://github.com/projectacrn/acrn-kernel and https://github.com/projectacrn/acrn to have those?

Currently, we just provide the flash and boot up test for the ACRN related patches. Currently ACRN based EVE is still not fully functional as XEN based EVE does, but we expect that it will have same test scope as XEN based EVE eventually.  Can EVE try to run the same test level of  Xen based Eve to ACRN-based EVE to figure what functionalities are missing?

File: pkg/grub/Dockerfile [hunk 2]
```diff
@@ -24,7 +24,7 @@ part_msdos cat echo test configfile"
 ENV GRUB_MODULES_x86_64="multiboot multiboot2 efi_uga"
 ENV GRUB_MODULES_aarch64="xen_boot"
 ENV GRUB_COMMIT=71f9e4ac44142af52c3fc1860436cf9e432bf764
-ENV GRUB_REPO=git://git.sv.gnu.org/grub.git
+ENV GRUB_REPO=https://git.savannah.gnu.org/git/grub.git
```
Comment (glguida) [inline pkg/grub/Dockerfile]: out of curiosity: any particular problem on using git.sv.gnu.org?

File: pkg/acrn/grub-hv.cfg [hunk 3]
```diff
@@ -1,4 +1,4 @@
 # Change hypervisor to ACRN
 
 set_global hv_boot_cmd "multiboot --quirk-modules-after-kernel /boot/acrn.32.out uart=bdf@0:18.1"
-set_global hv_linux_cmd "module $linux Linux_bzImage"
+set_global hv_linux_cmd "module $linux Linux_bzImage root=/dev/mmcblk0p2 textrw rootwait console=tty0 console=ttyS0 earlyprintk=ttyS0 consoleblank=0 no_timer_check quiet loglevel=3 i915.nuclear_pageflip=1 i915.avail_planes_per_pipe=0x01010F i915.domain_plane_owners=0x011111110000 i915.enable_gvt=1 hvlog=2M@0x5de00000 memmap=0x200000\$0x5de00000 memmap=0x400000\$0x5da00000 ramoops.mem_address=0x5da00000 ramoops.mem_size=0x400000 ramoops.console_size=0x200000 reboot_panic=p,w module_blacklist=dwc3_pci i915.enable_guc=0x02 cma=64M@0-"
```
Comment (glguida) [inline pkg/acrn/grub-hv.cfg]: Does it already work if you put $root here instead of hardwiring it to /dev/mmcblk0p2? Grub configuration should fill it up with the correct partition to boot.

Comment (jren1) [reply pkg/acrn/grub-hv.cfg]: Good idea, let me try to use $root to replace /dev/mmclblk0p2 to see if it works. Will try it tomorrow.

Comment (jren1) [reply pkg/acrn/grub-hv.cfg]: Veried with "root=$rootfs_root" and it works. I have updated the patches, please review.

**Comment by glguida** [comment]
+1 from me. This changes works and I can finally play around in a UP2 board.

**Comment by jren1** [comment]
Excellent!
